core/types/transactions/message.go

```diff
@@ -27,6 +27,9 @@ type CallMessage struct {
 
 	// Sender is the public key of the sender
 	Sender types.HexBytes `json:"sender"`
+
+	// Challenge is the signed challenge to prove the sender's identity
+	Challenge *types.SignedChallenge `json:"challenge"`
```

Comment:
Dang, I'm not really uncertain about signed challenge being part of the call request body or the JSON RPC service at all.  The *server*, not the RPC *service* may be the place to handle authentication. 

Specifically, in `(*Server) handlerV1`, we are dealing with the HTTP request, checking headers like http Basic auth (for admin server), then unmarshalling the request, and dispatching to the registered method handler.  In the server we could be issuing the challenges, and in this *server* method, we could be getting the the auth data from an http header, verifying it, and then calling into the service methods with the authenticated sender in the context.

The goal is to keep the authentication as far from the business of the *service* methods as possible.

I gotta think about it some more.

Reply:
@Yaiba @charithabandi and I just talked through a few related points:

- a template that includes (a) the challenge, (b) plain text explaining purpose and making it obvious it is not a transaction, and (c) and perhaps what json-rpc method it is intended for -- sign the template -- server generates the exepected template on the fly in order to verify the signuature
- segregating authentication API data from application data.  Namely, the server handling authentication while application/service method only needs the call data (schema method and action inputs etc).  I think @Yaiba has a good perspective on this with kgw.  AFAICT (correct me if I'm wrong) this is a close analogy but in this case we don't have a session cookie, instead a one-time authentication challenge that is discarded by server after use.  In kgw we didn't need to add the auth details the challenge to the `CallMessage`, and I'm not convinced we do now either.  Is private mode somehow conceptually more part of the application to the point that the service methods or engine need to be aware?  I don't think so.

Regarding the second point above, another consideration is if we change the authentication scheme, we would would have to change the core types.  Separating these pieces of information would provide flexibility.  Further, imagine if we devised a way to have transport layer authentication (e.g. mTLS with custom certification creation and verification to serve the same purpose), it would entirely vanish from the request body.